### PR TITLE
scripted-diff: Move util files to separate directory.

### PR DESCRIPTION
**Pull request by nuttycom**
Backport of bitcoin/bitcoin#14555

See the scripted diff in the commit.

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK

**Comment by daira** [comment]
Note: scripts should be in the commit they apply to, not the message that goes into the merge commit. I reviewed the latter before realising the former is what would be checked.

**Comment by nuttycom** [comment]
> Note: scripts should be in the commit they apply to, not the message that goes into the merge commit. I reviewed the latter before realising the former is what would be checked.

Yeah, the script *is* in the commit, it just got automatically copied into the PR since this is a single-commit PR. Thanks for editing the top post.

File: src/Makefile.am [hunk 1]
```diff
@@ -283,12 +283,12 @@ BITCOIN_CORE_H = \
   uint256.h \
   uint252.h \
   undo.h \
-  util.h \
+  util/system.h \
   util/match.h \
```
Comment (str4d) [inline src/Makefile.am:287]: Yay, friends!

File: src/Makefile.am [hunk 2]
```diff
@@ -283,12 +283,12 @@ BITCOIN_CORE_H = \
...
-  utilmoneystr.h \
-  utilstrencodings.h \
+  util/moneystr.h \
+  util/strencodings.h \
   utiltest.h \
```
Comment (str4d) [inline src/Makefile.am:290]: Was `utiltest.h` introduced by us, or by upstream in a different PR, or? Doesn't necessarily need to be done in this PR, but might be nice as an additional commit - if it was introduced by us. If it was from upstream, we should figure out why it wasn't handled in the PR being backported here.

**Review by str4d — APPROVED** [review]
utACK 9a7e2c153db20e7290a35b5e7a40e287dc9868a6 with non-blocking question.

**Review by str4d — APPROVED** [review]
utACK 71b6a59ec3f6ad41aba3ec1f5dc4bb421e832ec5

**Pull request merged**